identity-service/src/utils/withdrawUSDCInstructionsHelpers.js

```diff
@@ -0,0 +1,94 @@
+const {
+  ASSOCIATED_TOKEN_PROGRAM_ID,
+  TOKEN_PROGRAM_ID
+} = require('@solana/spl-token')
+const { Keypair } = require('@solana/web3.js')
+const config = require('../config')
+const { isRelayAllowedProgram, getInstructionEnum } = require('./relayUtils')
+
+const CLOSE_ACCOUNT_INSTRUCTION = 9
+const FEE_PAYER_ACCOUNT_INDEX = 1
+const CREATE_TOKEN_ACCOUNT_INSTRUCTION_INDEX = 0
+const USER_BANK_TRANSFER_INDEX_START = 1
+const USER_BANK_TRANSFER_INDEX_END = 3
+const ACCOUNT_TO_CREATE_INDEX = 1
+const ACCOUNT_TO_CLOSE_INDEX = 0
+
+/**
+ * Checks that the instruction is a create account instruction for an associated token account.
+ * @param {Instruction} instruction to check
+ * @returns true if validation passes
+ */
+const checkCreateAccountInstruction = (instruction) => {
+  return (
+    instruction.programId === ASSOCIATED_TOKEN_PROGRAM_ID.toString() &&
+    getInstructionEnum(instruction) === null
```

Comment:
why `null`?

Reply:
it appears that create account instructions have empty data. check out the first instruction in this tx: https://explorer.solana.com/tx/3bWxSri6jSvh3xxiMmKEg9XwY3q9i6Gjen2F3CXYzmV6LuZ7kxnb1eUVUEyyLw9dFcrabtwC27yKLrVGBpt9FkMW#ix-1

Reply:
contradicts that link you sent earlier tho, investigating.

Reply:
oh the link you posted is a different library SolanaProgramLibrary: https://github.com/solana-labs/solana-program-library/blob/fd479ffe6e016bf53cb4523e28ac88a0f83ccc3b/token/js/src/instructions/types.ts

Reply:
wdyt? probably not worth installing the lib just for that map?

Reply:
we have investigated and although the solana source code seems to imply that data is expected, we found multiple transactions in the wild that clearly don't: https://explorer.solana.com/tx/3pHqKnY3UmC5dR4kvnqKjufXiZnxXqTXWzocKtNFoacBxV73iD89MyTPwfhJy5NqCWcWRJnJxtHuWpyX8bzhYyKV